pkg/detectors/privatekey/ssh.go

```diff
@@ -91,6 +102,13 @@ func sshDialWithContext(ctx context.Context, network, addr string, config *ssh.C
 		return nil, fmt.Errorf("error dialing %s: %w", addr, err)
 	}
 
+	// Monitor the context and close the connection if canceled.
+	// This ensures that the connection is closed when the context is canceled, preventing leaks.
+	go func() {
+		<-ctx.Done()
+		conn.Close()
+	}()
```

Comment:
I'm confused why we need to spawn a goroutine to do this. Wouldn't the connection get closed when we're done using the client?

Also, is it an error to double-close the connection? We might be doing that if we fail to create on line 114.

Reply:
My mistake, I forgot I added logic to handle this in the caller.

Reply:
Correct, but we never finish using the client which keeps the conneciton open. The goroutine was added because the SSH client doesn't monitor the context after the connection is established. Without the goroutine, if the context is canceled after dialing, the client remains active and never closes the connection, leading to the hanging behavior. By spawning a goroutine that listens for context cancellation and closes the connection, we ensure that the SSH client respects the context's lifecycle beyond just the dialing phase. Maybe there is a better way to handle this?

The double close is defs an issue, thanks.

Reply:
I've defaulted to the wg.Wait() since both of you opted for the simpler approach. 👍